lib/package-managers/npm.js

```diff
@@ -44,6 +44,30 @@ function parseJson(result, data) {
     return json;
 }
 
+/**
+ * Check if package author changed between current and next version.
+ * @param {string} packageName Name of the package
+ * @param {string} currentVersion
+ * @param {string} latestVersion
+ * @returns {Promise<boolean>} A promise that fullfills with boolean value.
+ */
+async function packageAuthorChanged(packageName, currentVersion, latestVersion) {
+    npmConfig.fullMetadata = true;
+
+    const current = versionUtil.isWildCard(currentVersion.charAt(0)) ? currentVersion.substr(1) : currentVersion;
```

Comment:
What if `^5.0.0` is specified and the current version is `4.9.0`? It won't find anything at `result.versions['5.0.0']`.

`currentVersion` is any semver string, while `latestVersion` is a concrete version. I think you want to get the latest concrete version within `currentVersion`.

Reply:
Sorry, I didn't understood what you mean.

Reply:
Yeah, doesn't make sense to me either now that I read it :).

I think you are mixing up semver ranges with semver versions. If `currentVersion` specifies a version range, say, `^5.0.0`, then `current` will be set to `5.0.0`. However, there is not necessarily an existing version that corresponds to `5.0.0`. The latest may be `4.9.0`. So later  `result.versions[current]` may return `undefined`.

I don't think you can index into `result.versions` with a version range. Versions can only satisfy a version range.

Reply:
Ah, I see what you mean, I should try to fetch the latest version by using the semver utils instead of a fixed string which, in this case, would not exist as version.

Not sure how to tackle that thing if something like that happens.

Reply:
I can't offer a solution as I am working on other things, but I'm happy to review anything that you come up with.

Reply:
I got lazy these months but will look at it this weekend 😆

Reply:
I made a slighty refactor for this code block and tested on replit: [packageAuthorChanged test](https://repl.it/repls/SiennaWiseLivecd). I think this should cover those cases based on the semver ranges for both `current` and `latest`.

Reply:
Thanks! I will review.